pkgs/applications/networking/browsers/brave/default.nix

```diff
@@ -83,7 +85,9 @@ in stdenv.mkDerivation rec {
     dontBuild = true;
     dontPatchELF = true;
 
-    nativeBuildInputs = [ dpkg ];
+    nativeBuildInputs = [ dpkg wrapGAppsHook ];
+
+    buildInputs = [ glib gnome3.gsettings_desktop_schemas ];
```

Comment:
I think you'll need `gnome3.defaultIconTheme` too to have these icons in `XDG_DATA_DIRS`